hack/make-rules/helpers/cache_go_dirs.sh

```diff
@@ -62,8 +62,8 @@ if [[ -f "${CACHE}" ]]; then
     N=$(kfind -type d -newer "${CACHE}" -print -quit | wc -l)
     [[ "${N}" == 0 ]] && NEED_FIND=false
 fi
-mkdir -p $(dirname "${CACHE}")
-if $("${NEED_FIND}"); then
+mkdir -p "$(dirname "${CACHE}")"
+if ${NEED_FIND}; then
```

Comment:
Why did it change and what did it fix:
```
In ./hack/make-rules/helpers/cache_go_dirs.sh line 66:
if $("${NEED_FIND}"); then
   ^---------------^ SC2091: Remove surrounding $() to avoid executing output.
Ref: https://github.com/koalaman/shellcheck/wiki/SC2091
```
Example
**Problematic code**:
```
if $(which epstopdf)
then
  echo "Found epstopdf"
fi
```
**Correct code**:
```
if which epstopdf
then
  echo "Found epstopdf"
fi
```